### PR TITLE
Don't add empty accounts to bloom

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: ethcore/src/state/mod.rs [hunk 1]
```diff
@@ -340,18 +340,18 @@ impl State {
 
 	/// Determine whether an account exists.
 	pub fn exists(&self, a: &Address) -> bool {
-		self.ensure_cached(a, RequireCache::None, |a| a.is_some())
+		self.ensure_cached(a, RequireCache::None, false, |a| a.is_some())
```
Comment (rphmeier) [inline ethcore/src/state/mod.rs]: could use a comment here explaining why this is consensus-critical/ensure that tests will catch it if changed.

Comment (arkpar) [inline ethcore/src/state/mod.rs]: done

File: ethcore/res/ethereum/tests [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 862b4e3d4a9a7141af1b4aaf7dfe228a6a294614
+Subproject commit 97066e40ccd061f727deb5cd860e4d9135aa2551
```
Comment (tomusdrw) [inline ethcore/res/ethereum/tests]: Shouldn't be changed most probably.

Comment (rphmeier) [inline ethcore/res/ethereum/tests]: `git submodule update --init`?

**Pull request merged**